backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -371,13 +374,24 @@ object Compiler {
     def getCompilationOptions(inputs: CompileInputs): CompileOptions = {
       val sources = inputs.sources // Sources are all files
       val classpath = inputs.classpath.map(_.toFile)
+      val optionsWithoutFatalWarnings = inputs.scalacOptions.flatMap { option =>
```

Comment:
why not just a filter?
```scala
inputs.scalacOptions.filter {
  case "-Xfatal-warnings" =>
    isFatalWarningsEnabled = true
    false
  case _ =>
    true
}
```

Reply:
`filter` is implemented in terms of `flatMap` so it's one less level of indirection